src/app/util/ember-compatibility-functions.cpp

```diff
@@ -252,6 +252,13 @@ CHIP_ERROR ReadSingleClusterData(const ConcreteAttributePath & aPath, TLV::TLVWr
         ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), uint16_data));
         break;
     }
+    case ZCL_INT24U_ATTRIBUTE_TYPE: // Unsigned 24-bit integer
+    {
+        TLV::uint24_t uint24_data;
+        memcpy(static_cast<void *>(&uint24_data), attributeData, sizeof(uint24_data));
```

Comment:
Have we double-checked that the actual space in the attribute store that is allocated for this type is 3 bytes?

Reply:
I have found the default size in the array `GENERATED_DEFAULTS` in the autogenerated file: `endpoint_config.h` to be 3 bytes for both the Power and the LifetimeRunningHours attributes, which are both UINT24.

```
            /* Endpoint: 1, Cluster: Pump Configuration and Control (server), big-endian */                                        \
                                                                                                                                   \
            /* 2351 - LifetimeRunningHours, */                                                                                     \
            0x00, 0x00, 0x00,                                                                                                      \
                                                                                                                                   \
            /* 2354 - Power, */                                                                                                    \
            0x00, 0x00, 0x00,                                                                                                      \
```

And in the same file, I found the Attribute size, in the `GENERATED_ATTRIBUTES` array, to be 3 for those attributes as well.

```
            { 0x0015, ZAP_TYPE(INT24U), 3, ZAP_ATTRIBUTE_MASK(WRITABLE),                                                           \
              ZAP_LONG_DEFAULTS_INDEX(2351) }, /* LifetimeRunningHours */                                                          \
            { 0x0016, ZAP_TYPE(INT24U), 3, ZAP_ATTRIBUTE_MASK(WRITABLE), ZAP_LONG_DEFAULTS_INDEX(2354) }, /* Power */              \
```

Reply:
And if these values are the foundation for the attribute store, then I take it that they are actually 3 bytes long. But perhaps I'm not looking the correct places?